metricsbp/statsd.go

```diff
@@ -135,12 +135,21 @@ func (st Statsd) Counter(name string) metrics.Counter {
 	return st.Statsd.NewCounter(name, st.sampleRate)
 }
 
-// Histogram returns a histogram metrics to the name.
+// Histogram returns a histogram metrics to the name with no specific unit.
 //
 // It uses the DefaultSampleRate used to create Statsd object.
 // If you need a different sample rate,
-// you could use st.Statsd.NewTiming instead.
+// you could use st.Statsd.NewHistogram instead.
 func (st Statsd) Histogram(name string) metrics.Histogram {
+	return st.Statsd.NewHistogram(name, st.sampleRate)
+}
+
+// Timing returns a histogram metrics to the name with milliseconds as the unit.
+//
+// It uses the DefaultSampleRate used to create Statsd object.
+// If you need a different sample rate,
+// you could use st.Statsd.NewTiming instead.
+func (st Statsd) Timing(name string) metrics.Histogram {
```

Comment:
Should we use this rather than a Histogram in `Timer`?

Reply:
They both return the same type, after returned there's no way we can tell which way it was created before.

Reply:
Right, but it’s changing the suffix on the timer, doesn’t what it currently uses (ms) make more sense in the context of a timer?

Reply:
Yes, but it requires a lot of extra efforts to block people from using a normal histogram in a timer, which is not worthing it.

Reply:
Even if someone accidentally used Histogram in Timer, in the end they are still able to see all the p99/p90/p50 stuff, the only thing is that they need to manually set the unit on the graphs. That's not a big deal.

Reply:
Oh right, I was thinking that we created the Histogram in “NewTimer”